### ISSUE TITLE
Remove stupid hacky methods

### ISSUE THREAD

**Issue by steve-kaufman** [gomagedon/expectate#14]
Remove basically all methods except ToBe() and ToEqual(). The extremely hacky implementations of ToBeZero() and ToBeNil() are not worth cluttering the API. It's just as easy to write ToBe(0) and ToBe(nil). 

Yes, there are issues with comparing a float or some specific size int to the regular int 0, but we'll cross that bridge after we work out ToBe(), ToEqual(), NotToBe(), and NotToEqual(). Besides, if I'm trying to see if a float32 is 0, it's as simple as:
```golang 
expect(myFloat32).toBe(float32(0)) 
``` 
The case is the same for nil.

**In short, these functions are not idiomatic, and they should be removed.**

As for ToBeEmptyStr(), it's absolutely stupid, because it's longer than writing ToBe("").
